tokio/src/task/task_local.rs

```diff
@@ -277,6 +277,20 @@ impl<T: Copy + 'static> LocalKey<T> {
     }
 }
 
+#[allow(clippy::should_implement_trait)]
+impl<T: Clone + 'static> LocalKey<T> {
+    /// Returns a clone of the task-local value
+    /// if the task-local value implements `Clone`.
+    ///
+    /// # Panics
+    ///
+    /// This function will panic if the task local doesn't have a value set.
+    #[track_caller]
+    pub fn clone(&'static self) -> T {
```

Comment:
I'm slightly inclined to name this method as `clone_inner`, because naming it `clone` sound like it's an implementation of Clone for LocalKey.

Reply:
SGTM, amended.

Reply:
@fenollp 
Sorry for the confusion, but as Alice suggest [here](https://github.com/tokio-rs/tokio/pull/6433#issuecomment-2028023647), just adding a `Clone` bound to the existing `get` function seems sufficient here.

Reply:
Hmm, but if `T` only implements `Clone` then perhaps we still can't use `get`... 

@Darksonn 
Am I understanding your thoughts correctly?

Reply:
Why not? I can't immediately see what would break if we just change `get` to `T: Clone`.

Reply:
Ah, you meant to *replace* the bound from `Copy` to `Clone`? I misunderstood, sorry.

Then yes, I also don't see any issue.

Reply:
This seems to compile locally:
```diff
diff --git a/tokio/src/task/task_local.rs b/tokio/src/task/task_local.rs
index ba58ea6a..cb9d22c6 100644
--- a/tokio/src/task/task_local.rs
+++ b/tokio/src/task/task_local.rs
@@ -264,16 +264,16 @@ impl<T: 'static> LocalKey<T> {
     }
 }
 
-impl<T: Copy + 'static> LocalKey<T> {
+impl<T: Clone + 'static> LocalKey<T> {
     /// Returns a copy of the task-local value
-    /// if the task-local value implements `Copy`.
+    /// if the task-local value implements `Clone`.
     ///
     /// # Panics
     ///
     /// This function will panic if the task local doesn't have a value set.
     #[track_caller]
     pub fn get(&'static self) -> T {
-        self.with(|v| *v)
+        self.with(|v| v.clone())
     }
 }
```

Reply:
Guys. Make up your mind, this is exactly my original commit.

Reply:
What we want to do is [this change](https://github.com/tokio-rs/tokio/pull/6433#discussion_r1555673348). 
Sorry if I missing something, but I guess you were originally trying to add a [new method with a Clone boundary](https://github.com/tokio-rs/tokio/commit/b6c21ad09363a21533919b50af7c4938551de9b2)?

Reply:
Is it? If so, then you changed it before I got a chance to see it, because the thing I saw originally added a new method called `clone` instead of changing the existing `get` method.

Either way, I have made up my mind. I would like to change `get` instead of introducing a new method.

Reply:
I'm not sure I understand what is going on in this review. Who's authoritative (I see one member and one contributor)? Why remove `get` for `Copy` inner types? Why not just add a similar impl for `Clone`? These are two orthogonal things.

I'm making this PR to add a feature with a simpler dev UX than right now. I wasn't planning on removing anything. I couldn't just extend on `get` for `Copy + Clone` simply because my type isn't `Copy` (it's an `Arc<_>`).

I'm not sure how to move forward anymore. Please help :)

Reply:
Thank you very much for the clarification! PR amended.